This backported patch avoids the upstream scons check to prevent unintended
issues and can be removed next release bump where it may be preferable to
use the newer cmake build system.

https://bugs.gentoo.org/906123
https://bugs.gentoo.org/903001
https://github.com/apache/serf/commit/df0d2d0dbdf88576f26da9c71df3ab6249d351dc

From df0d2d0dbdf88576f26da9c71df3ab6249d351dc Mon Sep 17 00:00:00 2001
From: Stefan Sperling <stsp@apache.org>
Date: Tue, 20 Dec 2016 09:57:08 +0000
Subject: [PATCH] Fix another build problem with LibreSSL.

* SConstruct: Check for OpenSSL_malloc_init() and provide the
   SERF_HAVE_OPENSSL_MALLOC_INIT feature flag.

* buckets/ssl_buckets.c
  (init_ssl_libraries): Use SERF_HAVE_OPENSSL_MALLOC_INIT instead
   of USE_LEGACY_OPENSSL.

git-svn-id: https://svn.apache.org/repos/asf/serf/trunk@1775242 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/buckets/ssl_buckets.c b/buckets/ssl_buckets.c
index b01e535..27c84c7 100644
--- a/buckets/ssl_buckets.c
+++ b/buckets/ssl_buckets.c
@@ -1156,7 +1156,7 @@ static void init_ssl_libraries(void)
         }
 #endif
 
-#ifdef USE_OPENSSL_1_1_API
+#ifndef LIBRESSL_VERSION_NUMBER
         OPENSSL_malloc_init();
 #else
         CRYPTO_malloc_init();
